### PR TITLE
Clarify the definition of "index"

### PR DESCRIPTION
**Pull request by littledan**
Addresses part of #59

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -106,3 +106,7 @@ It is easy to create a stateless API based on this stateful one, or vice versa,
 Q: Why is this an Intl API instead of String methods?
 
 A: All of these break types are actually locale-dependent, and some allow complex options. The result of the `segment` method is a SegmentIterator. For many non-trivial cases like this, analogous APIs are put in ECMA-402's Intl object. This allows for the work that happens on each instantiation to be shared, improving performance. We could make a convenience method on String as a follow-on proposal.
+
+Q: What exactly does the index refer to?
+
+An index *n* refers to the boundary preceding the code unit *n*. For example, when iterating over the string "Hello, world💙" by words in English, there will be boundaries at 0, 5, 6, 7, 12, and 1. The definition of these boundary indices does not depend on whether forwards or backwards iteration is used.
```
Comment (gibson042) [inline README.md]:
```suggestion
An index *n* refers to the boundary preceding the code unit *n*. For example, when iterating over the string "Hello, world💙" by words in English, there will be boundaries at 0, 5, 6, 7, 12, and 13. The definition of these boundary indices does not depend on whether forwards or backwards iteration is used.
```

File: spec.html [hunk 2]
```diff
@@ -249,9 +249,9 @@ <h1>AdvanceSegmentIterator ( _iterator_, _direction_ )</h1>
         1. Let _index_ be _iterator_.[[SegmentIteratorIndex]].
         1. Let _len_ be the length of _string_.
         1. If _direction_ is ~forwards~ and _index_ is _len_, or if _direction_ is ~backwards~ and _index_ is *0*, return *true*.
-        1. Find the next or previous (based on _direction_) break in _string_ from the code unit index _index_ based on the locale, and granularity in _segmenter_.[[Locale]], _segmenter_.[[SegmenterGranularity]].
+        1. Find the next or previous (based on _direction_) boundary in _string_ from the code unit index _index_ based on the locale, and granularity in _segmenter_.[[Locale]], _segmenter_.[[SegmenterGranularity]].
```
Comment (gibson042) [inline spec.html]:
```suggestion
        1. Find the next or previous (based on _direction_) boundary in _string_ from the boundary immediately preceding code unit index _index_ based on the locale, and granularity in _segmenter_.[[Locale]], _segmenter_.[[SegmenterGranularity]].
```

File: README.md (hunk 1 above)
Comment (littledan) [reply README.md]: This should actually say 14 (I used a blue, not red, heart to draw out the difference between code point and code unit)

File: spec.html (hunk 2 above)
Comment (littledan) [reply spec.html]: Perfect, thanks!

**Pull request merged**